carma_wm/src/collision_detection.cpp

```diff
@@ -0,0 +1,179 @@
+                yaw = std::acos(vehicle_vector.dot(x_axis)/(vehicle_vector.norm() * x_axis.norm()));
+
+                tf2::Quaternion orientation;
+                orientation.setRPY(0, 0, yaw);
+
+                geometry_msgs::Pose pose;
+                pose.position.x = tp.trajectory_points[i].x;
+                pose.position.y = tp.trajectory_points[i].y;
+
+                pose.orientation.x = orientation.getX();
+                pose.orientation.y = orientation.getY();
+                pose.orientation.z = orientation.getZ();
+                pose.orientation.w = orientation.getW();
+
+                v.future_polygons.push_back(CollisionChecking::ObjectToBoostPolygon<polygon_t>(pose, size));
+            }
+
+            return v;
+        };
+
+        bool CollisionChecking::DetectCollision(collision_detection::MovingObject ob_1, collision_detection::MovingObject ob_2, int target_time) {            
+
+            collision_detection::MovingObject ob_1_after = CollisionChecking::PredictObjectPosition(ob_1,target_time);
+            collision_detection::MovingObject ob_2_after = CollisionChecking::PredictObjectPosition(ob_2,target_time);
+
+            if(CollisionChecking::CheckPolygonIntersection(ob_1_after, ob_2_after)){
+                return true;
+            }
+            
+            return false;
+        };
+
+        bool CollisionChecking::CheckPolygonIntersection(collision_detection::MovingObject ob_1, collision_detection::MovingObject ob_2) {    
+
+                std::deque<polygon_t> output;
+
+                boost::geometry::intersection(ob_1.object_polygon, ob_2.object_polygon, output); 
+
+                BOOST_FOREACH(polygon_t const& p, output)
+                {
+                    if(boost::geometry::area(p) > 0){
+                        return true;
+                    }
+                }
+
+            return false;
+        };
+
+        collision_detection::MovingObject CollisionChecking::PredictObjectPosition(collision_detection::MovingObject op, int target_time){
+
+            int size = 0;
+            for(int i = 0; i< target_time; i++){
+                size = size + op.future_polygons[i].outer().size();
+            }
+
+            std::vector<point_t> unioin_polygon_points;
+
+            unioin_polygon_points.reserve(size); 
+
+            for(int i = 0; i< target_time; i++){
+                unioin_polygon_points.insert( unioin_polygon_points.end(), op.future_polygons[i].outer().begin(), op.future_polygons[i].outer().end());
+            }
+
+            polygon_t union_polygon;  
+            boost::geometry::assign_points(union_polygon, unioin_polygon_points);
+
+            polygon_t hull_polygon;
+            boost::geometry::convex_hull(union_polygon, hull_polygon);
+
+            collision_detection::MovingObject output_object = {hull_polygon, op.linear_velocity};
+            
+            return output_object;
+        };
```

Comment:
This function needs some comments describing what is happening. Is it creating a convex hull around all the predictions?

Reply:
yes i added comments in header file